syntax = "proto3";

import "farm_ng/core/resource.proto";
import "farm_ng/perception/camera_model.proto";
import "google/protobuf/wrappers.proto";

package farm_ng.perception;
option go_package = "github.com/farm-ng/genproto/perception";

message Image {
  // The associated image or video resource
  farm_ng.core.Resource resource = 1;
  CameraModel camera_model = 2;

  // If resource is a video, a frame number offset into the video file
  // The first frame is frame 0
  google.protobuf.UInt32Value frame_number = 3;

  // If resource is a video, the frame rate of the video
  google.protobuf.DoubleValue fps = 4;
}
